etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestrator.java

```diff
@@ -15,45 +23,89 @@ public class PartnerMetadataOrchestrator {
     private static final PartnerMetadataOrchestrator INSTANCE = new PartnerMetadataOrchestrator();
 
     @Inject PartnerMetadataStorage partnerMetadataStorage;
+    @Inject ReportStreamEndpointClient rsclient;
+    @Inject Formatter formatter;
 
     public static PartnerMetadataOrchestrator getInstance() {
         return INSTANCE;
     }
 
     private PartnerMetadataOrchestrator() {}
 
-    public void updateMetadataForReceivedOrder(String submissionId, Order<?> order)
+    public void updateMetadataForReceivedOrder(String receivedSubmissionId, Order<?> order)
             throws PartnerMetadataException {
-        // will call the RS history API given the submissionId (albeit, right now this won't work
-        // given the way RS works).
-        // from the history API response, extract the sender (organization + sender client), and
-        // time received.
+        // will call the RS history API given the submissionId, currently blocked by:
+        // https://github.com/CDCgov/prime-reportstream/issues/12624
+        // from the response, extract the "sender" and "timestamp" (timeReceived)
         // we will calculate the hash.
         // then we call the metadata storage to save this stuff.
 
+        String sender = "unknown";
+        Instant timeReceived = Instant.now();
+        String hash = String.valueOf(order.hashCode());
         PartnerMetadata partnerMetadata =
-                new PartnerMetadata(
-                        submissionId, "senderName", "receiverName", Instant.now(), "abcd");
+                new PartnerMetadata(receivedSubmissionId, sender, timeReceived, hash);
         partnerMetadataStorage.saveMetadata(partnerMetadata);
     }
 
-    public void updateMetadataForSentOrder(
-            String receivedSubmissionId, String sentSubmissionId, Order<?> order)
+    public void updateMetadataForSentOrder(String receivedSubmissionId, String sentSubmissionId)
             throws PartnerMetadataException {
-        // call the metadata storage and add the sent order's submission ID to the existing metadata
-        // entry
-        // PartnerMetadata may need to be updated to store both the received order's submission ID
-        // _and_ the sent order's submission ID.
+
+        PartnerMetadata partnerMetadata =
+                partnerMetadataStorage.readMetadata(receivedSubmissionId).orElseThrow();
+        if (!Objects.equals(partnerMetadata.sentSubmissionId(), sentSubmissionId)) {
```

Comment:
You should be able to refactor this into something like 

```suggestion
        if (sentSubmissionId != null && !sentSubmissionId.equals`(partnerMetadata.sentSubmissionId()) {
```

Reply:
Thanks, I'll use your suggestion to compare. We don't need to check sentSubmissionId for null since we check in `saveSentOrderSubmissionId`.